applications/FluidDynamicsApplication/symbolic_generation/navier_stokes/generate_navier_stokes_element.py

```diff
@@ -104,28 +118,38 @@
     v_gauss = v.transpose()*N
 
     ## Convective velocity definition
-    if (linearisation == "Picard"):
-        vconv = DefineMatrix('vconv',nnodes,dim)    # Convective velocity defined a symbol
-    elif (linearisation == "FullNR"):
-        vmesh = DefineMatrix('vmesh',nnodes,dim)    # Mesh velocity
-        vconv = v - vmesh                           # Convective velocity defined as a velocity dependent variable
-
-    vconv_gauss = vconv.transpose()*N
+    if convective_term:
+        if (linearisation == "Picard"):
+            vconv = DefineMatrix('vconv',nnodes,dim)    # Convective velocity defined a symbol
+        elif (linearisation == "FullNR"):
+            vmesh = DefineMatrix('vmesh',nnodes,dim)    # Mesh velocity
+            vconv = v - vmesh                           # Convective velocity defined as a velocity dependent variable
+        else:
+            raise Exception("Wrong linearisation \'" + linearisation + "\' selected. Available options are \'Picard\' and \'FullNR\'.")
+        vconv_gauss = vconv.transpose()*N
 
     ## Compute the stabilization parameters
-    vconv_gauss_norm = 0.0
-    for i in range(0, dim):
-        vconv_gauss_norm += vconv_gauss[i]**2
-    vconv_gauss_norm = sqrt(vconv_gauss_norm)
+    stab_norm_v = 0.0
+    if convective_term:
+        for i in range(0, dim):
+            stab_norm_v += vconv_gauss[i]**2
+    else:
+        # Note that in the Stokes case we define an auxiliary velocity for the stabilization constants calculation
+        # This is intentionally done to avoid introducing the norm of velocity in the residual linearisation
+        v_aux = DefineVector('v_aux', dim)
+        for i in range(0, dim):
+            stab_norm_v += v_aux[i]**2
+    stab_norm_v = sqrt(stab_norm_v)
 
-    tau1 = 1.0/((rho*dyn_tau)/dt + (stab_c2*rho*vconv_gauss_norm)/h + (stab_c1*mu)/(h*h))   # Stabilization parameter 1
-    tau2 = mu + (stab_c2*rho*vconv_gauss_norm*h)/stab_c1                                    # Stabilization parameter 2
+    tau1 = 1.0/((rho*dyn_tau)/dt + (stab_c2*rho*stab_norm_v)/h + (stab_c1*mu)/(h*h))   # Stabilization parameter 1
```

Comment:
also, if one aims to solve the steady stokes problem, should the contribution of the `dynamic_tau` term omitted from  `tau1`?

Reply:
Same as above. We are not using a steady implementation but deactivating the inertial terms.